internal/tofu/node_resource_abstract_instance.go

```diff
@@ -2779,8 +2779,9 @@ func (n *NodeAbstractResourceInstance) apply(
 	newVal := resp.NewState
 
 	// If we have paths to mark, mark those on this new value
-	if len(afterPaths) > 0 {
-		newVal = newVal.MarkWithPaths(afterPaths)
+	marks := combinePathValueMarks(afterPaths, schema.ValueMarks(newVal, nil))
+	if len(marks) > 0 {
+		newVal = newVal.MarkWithPaths(marks)
```

Comment:
```suggestion
	newValMarks := combinePathValueMarks(afterPaths, schema.ValueMarks(newVal, nil))
	if len(newValMarks) > 0 {
		newVal = newVal.MarkWithPaths(newValMarks)
```
Just to avoid collision with the package marks